lib/Sema/TypeCheckType.cpp

```diff
@@ -3730,6 +3730,24 @@ TypeResolver::resolveIdentifierType(IdentTypeRepr *IdType,
     return ErrorType::get(getASTContext());
   }
 
+    auto *dc = getDeclContext();
+    auto &ctx = getASTContext();
+
+    if ( ctx.LangOpts.hasFeature(Feature::ImplicitSome) & !options.is(TypeResolverContext::Inherited) ){
```

Comment:
Perhaps you want to check options.isConstraintImplicitExistential() instead of options.is(TypeResolverContext::Inherited)?